wire-ios-data-model/Source/Notifications/ObjectObserverTokens/Helpers/ClosureBasedUserChangeObserver.swift

```diff
@@ -0,0 +1,34 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+/// Allows for handling user change events via closure.
+@objc(ZMClosureBasedUserChangeObserver)
+public final class ClosureBasedUserChangeObserver: NSObject, UserObserver {
```

Comment:
I started here with the question: why does it need to inherit from `NSObject`? Can't `ClosureBasedUserChangeObserver` be a struct?

- The `protocol UserObserver: NSObjectProtocol` requires it, but is it really necessary?
- I replaced it with `protocol UserObserver: AnyObject` and the compiler doesn't complain.
- Then I removed it like `protocol UserObserver` and the compiler doesn't complain 

From my understanding the inheritance from `NSObjectProtocol` is required when the code requires a class implementing it and sometimes it's important to assume so. In this particular case I think it's not necessary and we can remove the constraint.

Reply:
I added it because the protocol required it, but you're right, we can just remove Obj-C support: 55ed881a7eb24642f8dee154f63aa63a56766770

Reply:
Oh I see because of `@objc` we need to have a class conformance. Thanks!